js/utils.js

```diff
@@ -190,10 +190,164 @@
 	};
 
 	/**
+	 * Build a helper function to retrieve related model.
+	 *
+	 * @param  {Backbone.Model} parentModel - The parent model.
+	 * @param  {int}    modelId - The model ID if the object to request
+	 * @param  {string} modelName - The model name to use when constructing the model.
+	 * @param  {string} embedSourcePoint - Where to check the embeds object for _embed data.
+	 * @param  {string} embedCheckField - Which model field to check to see if the model has data.
+	 *
+	 * @return {Deferred.promise}        A promise which resolves to the constructed model.
+	 */
+	wp.api.utils.buildModelGetter = function buildModelGetter( parentModel, modelId, modelName, embedSourcePoint, embedCheckField ) {
+		var getModel, embeddeds, attributes, deferred;
+
+		deferred  = jQuery.Deferred();
+		embeddeds = parentModel.get( '_embedded' ) || {};
+
+		// Verify that we have a valid object id.
+		if ( ! _.isNumber( modelId ) || 0 === modelId ) {
+			deferred.reject();
+			return deferred;
+		}
+
+		// If we have embedded object data, use that when constructing the getModel.
+		if ( embeddeds[ embedSourcePoint ] ) {
+			attributes = _.findWhere( embeddeds[ embedSourcePoint ], { id: modelId } );
+		}
+
+		// Otherwise use the modelId.
+		if ( ! attributes ) {
+			attributes = { id: modelId };
+		}
+
+		// Create the new getModel model.
+		getModel = new wp.api.models[ modelName ]( attributes );
+
+		// If we didn’t have an embedded getModel, fetch the getModel data.
+		if ( ! getModel.get( embedCheckField ) ) {
+			getModel.fetch( {
+				success: function( model ) {
+					var updatedEmbeddeds = _.clone( parentModel.get( '_embedded' ) || {} );
+					if ( ! updatedEmbeddeds[ embedSourcePoint ] ) {
```

Comment:
@westonruter can you explain why you are caching the parent embed here? does this prevent duplicate lookup calls? can you explain how to reproduce and i can work on a unit test to demonstrate what this fixes? thanks